backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/VmInfoBuildUtils.java

```diff
@@ -1677,4 +1677,48 @@ public static boolean isVmWithHighNumberOfX86Vcpus(VM vm) {
         return vm.getClusterArch().getFamily() == ArchitectureType.x86
                 && maxNumberOfVcpus(vm) >= VmCpuCountHelper.HIGH_NUMBER_OF_X86_VCPUS;
     }
+
+    /**
+     * Return a recommended TSEG size in MB or null to use the default value.
+     *
+     * Large VMs with UEFI will not start if they don't have TSEG large enough.
+     * There is no rule what "large" is, it's a VM with "too many" vCPUs or "too large" memory.
+     * There are just some suggestions:
+     * - To use 48 MB for VMs with many vCPUs or large memory:
+     *   https://bugzilla.redhat.com/show_bug.cgi?id=2074149#c22
+     * - The same in a different context:
+     *   https://bugzilla.redhat.com/show_bug.cgi?id=1469338#c30
+     * - To increase the size by unspecified amount for many vCPUs and by 8 MB of each TB of
+     *   address space (which is max RAM + NVDIMM size in our case):
+     *   https://bugzilla.redhat.com/show_bug.cgi?id=1469338#c7
+     * - Confirmation that 48 MB may not be enough for really large VMs:
+     *   https://bugzilla.redhat.com/show_bug.cgi?id=2074149#c28
+     *
+     * If UEFI is not used, we should be safe and use the default value.
+     *
+     * If the VM is small, we also use the default value, because the TSEG size is taken
+     * from the memory available to the guest.
+     *
+     * Otherwise, we start with the default value (16 MB currently) and add 24 MB
+     * (which implies at least the recommended value of 48 MB together with the default value
+     * and additional value per memory) for VMs with many vCPUs and 8 MB per each (even partial)
+     * TB of RAM. This is probably a bit wasteful but it's better than risking the VM won't start.
+     **/
+    public Integer tsegSizeMB(VM vm, MemoizingSupplier<Map<String, HostDevice>> hostDevicesSupplier) {
+        if (vm.getBiosType() == null || !vm.getBiosType().isOvmf()) {
+            return null;
+        }
+        final int cpuLimit = 128;
+        final int smallMemory = 16 * 1024;
```

Comment:
we seem to be the first to default to UEFI and so we face such issues and who knows how it really behaves.. what do you think about making this configurable (using config values) so it can be tuned if needed?